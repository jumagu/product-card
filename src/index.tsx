export * from './hooks';
export * from './components';
export * from './interfaces/product.interfaces';
