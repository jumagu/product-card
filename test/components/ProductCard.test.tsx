import React from 'react';
import renderer from 'react-test-renderer';

import { product1 } from '../data/products';
import { ProductCard } from '../../src/components';

const { act } = renderer;

describe('ProductImage', () => {
  test('Should render correctly with the custom image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => <h1>Product Card</h1>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Should increment the quantity', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {({ quantity, incrementBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{quantity}</span>
            <button onClick={() => incrementBy(1)}></button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('1');
  });
});
