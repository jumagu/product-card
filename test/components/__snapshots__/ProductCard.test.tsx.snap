// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Should increment the quantity 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductImage Should render correctly with the custom image 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;
